### ISSUE TITLE
broken hyperlinks referencing subset data

### ISSUE THREAD

**Issue by Mikejmnez** [OPENDAP/documentation#20]
The Constraint Expressions part of the Documentation has some broken links. These are very easy to fix, but I wanted to open this issue to:

1) Understand the broken links to make the proper modifications.
2) Raise an issue so that a PR can be connected to it.

### With regards to 1)
Below is one of the broken links in the documentation.  The broken hyperlinks under `Constraint Expressions` all share the similar behavior. The hyperlink refers to the test data `http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz?sst[3][4][5]` (DAP2)
https://github.com/OPENDAP/documentation/blob/6ab6ebb0e13654b9cba9f54209133f8c1b7460ee/UserGuideComprehensive.html#L2335

 The url has already been escaped, but the link fails to open on the browser with a header that says 
```python
Hyrax - Resource Not Found (404). 
```
I noticed that the file type is `asc` (non-structure ASCII). Without the `asc` the hyper link downloads the data.

a) Is the goal to open an ASCII type file on the browser to visualize the stream of numbers?
b) Is the goal to download data (any format)?
c) Should the proper link be a DDS, something like [this](http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz.dds?sst%5B3%5D%5B4%5D%5B5%5D)?

Whether a, b or c), will inform me on how to proceed to fix these hyperlinks and submit a PR.

**Comment by Mikejmnez** [comment]
In addition to some broken hyperlinks on the documentation, the README file references the [https://opendap.github.io/](https://opendap.github.io/) as the url for the documentations. However, this address is wrong and will result in a 404: Site note found.

Instead, the README should have the correct url: [https://opendap.github.io/documentation](https://opendap.github.io/documentation).

A PR that fixes the broken URLs on the `Constraint Expression` subsection of the documentation should also correct the README url

**Comment by jgallagher59701** [comment]
Regarding the Constraint expression issue above.

One way to look into those errors is to look at http://test.opendap.org/dap/data/nc/ and then find sst.mnmean.nc.gz (http://test.opendap.org/dap/data/nc/sst.mnmean.nc.gz.dmr.html)

I think that the new(er) server requires .ascii and won't accept .asc - annoying, I agree. I'll look around a bit more.

**Comment by jgallagher59701** [comment]
The new, DAP4 way to make that request is: http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz.dap.csv?dap4.ce=/sst[3][4][5]. The encoded URL, which you can get from the form interface, is: http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz.dap.csv?dap4.ce=/sst%5B3%5D%5B4%5D%5B5%5D

But...

This does not seem to be working at all correctly. It seems that many things that should work fail in one way or another.

**Comment by jgallagher59701** [comment]
Here's a summary of what _should_ be happening:

DAP2
To ask for an ASCII/CSV response, use '.ascii'

DAP4
To ask for an ASCII/CSV response, use '.dap.csv'

**Comment by Mikejmnez** [comment]
Thanks for the guidance! I will work to get the proper hyperlinks on the Documentation

**Comment by Mikejmnez** [comment]
## DAP2 works fine
Pasting this url: [http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz.ascii?sst[3][4][5]](http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz.ascii?sst[3][4][5]) on a browser shows:
```python
Dataset: sst.mnmean.nc
sst.lon, 10
sst.sst[sst.time=19813][sst.lat=80], -51
```
NOTE: No need to escape the characters.

### DAP4 not as expected
If I paste the dap4 url:  [http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz.dap.csv?dap4.ce=/sst[3][4][5]](http://test.opendap.org/opendap/data/nc/sst.mnmean.nc.gz.dap.csv?dap4.ce=/sst[3][4][5]) onto a browser it downloads the [sst.mnmean.nc.gz.csv](sst.mnmean.nc.gz.csv) file. Looking at its contents with a simple text editor

```python
Dataset: sst.mnmean.nc
/sst[0][0], 
```
With pandas:
 
<img width="350" alt="Screenshot 2024-04-24 at 9 13 45 PM" src="https://github.com/OPENDAP/documentation/assets/8241481/1d52350f-431a-49e7-a041-abcaff268d28">

That is pretty confusing...

**Comment by jgallagher59701** [comment]
That is a bug. Can you file that - as a BES issue in GitHub. The first step should probably be to update the test.o.o server to the latest version of the code.

**Comment by Mikejmnez** [comment]
Will do! 
Regarding the Documentation, what I will do for now is to fix the hyperlinks to match the DAP2 implementation. We can then later replace with the DAP4 approach. 
I want to trigger a build of the documentation again with a PR, to make sure things works fine now with regards to Travis ci.

**Comment by Mikejmnez** [comment]
closed by #21
